src/cljs/athens/devcards/devtool.cljs

```diff
@@ -0,0 +1,427 @@
+
+
+; TODO add truncation of long strings here
+(defn edn-viewer
+  [data _]
+  [:pre [:code (with-out-str (cljs.pprint/pprint data))]])
+
+
+(defn coll-viewer
+  [coll add-nav!]
+  [data-table ["idx" "value"]
+   (->> coll
+        (map-indexed (fn [idx item]
+                       (with-meta [idx item] {:row-value item})))
+        vec)
+   add-nav!])
+
+
+(defn map-viewer
+  [m add-nav!]
+  [data-table ["key" "value"]
+   (map (fn [[k v]] (with-meta [k v] {:row-value v})) m)
+   add-nav!])
+
+
+(defn maps-viewer
+  [ms add-nav!]
+  (let [headers (into ["idx"] (->> ms (mapcat keys) distinct))
+        rows (map-indexed (fn [idx m]
+                            (with-meta (into [idx]
+                                             (for [h (rest headers)] (get m h)))
+                              {:row-value m}))
+                          ms)]
+    [data-table headers rows add-nav!]))
+
+
+(defn tuples-viewer
+  [colls add-nav!]
+  (let [max-count (->> colls
+                       (map count)
+                       (apply max))
+        headers (into ["idx"] (range max-count))
+        rows (map-indexed (fn [idx coll]
+                            (with-meta (into [idx]
+                                             (for [i (range max-count)] (get coll i)))
+                              {:row-value coll})))]
+    [data-table headers rows add-nav!]))
+
+
+(defn associative-not-sequential?
+  [x]
+  (and (associative? x)
+       (not (sequential? x))))
+
+
+(defn sequence-of-maps?
+  [x]
+  (and (sequential? x)
+       (every? associative? x)
+       (not (every? sequential? x))))
+
+
+(defn tuples?
+  [x]
+  (and (sequential? x)
+       (every? sequential? x)))
+
+
+(def viewers
+  [{:athens.viewer/id :athens.browser/edn
+    :athens.viewer/pred (constantly true)
+    :athens.viewer/fn edn-viewer}
+   {:athens.viewer/id :athens.browser/coll
+    :athens.viewer/pred coll?
+    :athens.viewer/fn coll-viewer}
+   {:athens.viewer/id :athens.browser/map
+    :athens.viewer/pred associative-not-sequential?
+    :athens.viewer/fn map-viewer}
+   {:athens.viewer/id :athens.browser/maps
+    :athens.viewer/pred sequence-of-maps?
+    :athens.viewer/fn maps-viewer}])
+   ;{:athens.viewer/id :athens.browser/tuples
+   ; :athens.viewer/pred tuples?
+   ; :athens.viewer/fn tuples-viewer}])
+
+(def viewer-preference
+  [:athens.browser/maps
+   :athens.browser/map
+   :athens.browser/tuples
+   :athens.browser/coll
+   :athens.browser/edn])
+
+
+(defn applicable-viewers
+  [data]
+  (->> viewers
+       (filter (fn [{:keys [athens.viewer/pred]}] (pred data)))
+       (map :athens.viewer/id)
+       (sort-by #(.indexOf viewer-preference %))))
+
+
+(def indexed-viewers
+  (->> viewers
+       (map (juxt :athens.viewer/id identity))
+       (into {})))
+
+
+(defn dropdown
+  [options selected on-change]
+  [:select {:on-change on-change
+            :value selected}
+   (for [opt options]
+     ^{:key opt}
+     [:option {:value opt}
+      opt])])
```

Comment:
Personally, I think I'd like to have all the options shown at once, like a horizontal list of buttons. Then we can see the all the options without clicking on the dropdown, and can change with one click rather than two.

Thoughts @shanberg ?

https://jelmerderonde.github.io/athens/cards.html#!/athens.devcards.devtool

Reply:
I agree! Fixed it.